File: lib_core/src/main/java/river/chat/lib_core/net/retrofit/HeadInterceptor.java
package river.chat.lib_core.net.retrofit;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * Created by beiyongChao on 2023/2/20
 * Description:
 */
public class HeadInterceptor implements Interceptor {
        @Override
        public Response intercept(Chain chain) throws IOException {

                long timestamp = System.currentTimeMillis();

                Request request = chain.request()
                        .newBuilder()
//                        .header("Connection", "close")
                        .header("timestamp", timestamp + "")
                        .build();
                return chain.proceed(request);
        }
}

File: lib_core/src/main/java/river/chat/lib_core/net/interceptor/HeadInterceptor.kt
package river.chat.lib_core.net.interceptor


import okhttp3.Interceptor
import okhttp3.Request
import river.chat.lib_core.config.AppSystemConfigKey
import river.chat.lib_core.router.plugin.core.getPlugin
import river.chat.lib_core.router.plugin.module.UserPlugin
import river.chat.lib_core.utils.longan.appVersionName

/**
 * Created by beiyongChao on 2023/4/27
 * Description: 请求头 拦截器
 */


class HeadInterceptor : Interceptor {
    override fun intercept(chain: Interceptor.Chain): okhttp3.Response {
        val token: String = getPlugin<UserPlugin>().getUser()
            .token
        val request: Request = chain.request()
            .newBuilder() //                        .header("Connection", "close")
            .header("header_token", token)
            .header("header_lng", "")
            .header("header_lat", "")
            .header("header_source", AppSystemConfigKey.PLATFORM + "")
            .header("header_version", appVersionName)
            .build()

        return chain.proceed(request)
    }
}